src/nvim/terminal.c

```diff
@@ -172,6 +173,39 @@ void terminal_teardown(void)
   pmap_init(ptr_t, &invalidated_terminals);
 }
 
+static int on_osc_fallback(const char *command, size_t cmdlen, void *user)
+{
+  // Check for an OSC foreground/background color request, and respond
+  // accordingly.
+  if (cmdlen == 4) {
+    bool fg_request = strncmp(command, "10;?", 4) == 0;
+    bool bg_request = !fg_request && strncmp(command, "11;?", 4) == 0;
+    if (fg_request || bg_request) {
+      // WARN: This does not return the actual foreground/background color, but
+      // rather returns:
+      //   - fg=white/bg=black when Neovim option 'background' is 'dark'
+      //   - fg=black/bg=white when Neovim option 'background' is 'light'
+      char_u *bg_option;
+      long dummy;
+      get_option_value("background", &dummy, &bg_option, OPT_GLOBAL);
+      bool bg_option_dark = STRNCMP(bg_option, "dark", 4) == 0;
+      xfree(bg_option);
+      Terminal *term = user;
+      int red = 0x00, green = 0x00, blue = 0x00;
+      if ((fg_request && bg_option_dark) || (bg_request && !bg_option_dark)) {
+        red = green = blue = 0xff;
+      }
+      char data[25];
+      snprintf(
+          data, sizeof(data), "\x1b]%.3srgb:%02x%02x/%02x%02x/%02x%02x\x07",
+          command, red, red, green, green, blue, blue);
+      terminal_send(term, data, sizeof(data) - 1);  // excludes null byte
```

Comment:
You can use the `S_LEN` macro here

```suggestion
      terminal_send(term, S_LEN(data));
```

Reply:
Done (f30cc5376df62449956f9700afa903c689c7986b).